### ISSUE TITLE
Create The Trainer Module

### ISSUE THREAD

**Issue by jsphon** [jsphon/reinforcement_learning#51]
This might replace the learner module.

The problem:

The simple line world model does not converge for certain neural network configurations. Eyeballing the calculated target functions, and running the tests on them, suggests that they are correct.

I have an inkling that this is because the learning rate is affected by:

 - the learning rate of the optimiser, which trains the value function towards the target values.
 - the number of training operations per iteration.

The current implementation trains in these steps:

 1 - calculate the target values
 2 - train the value function on the targets

Step 2 currently only does 1 step. I think it might work better if it does a few steps.
